deepchecks/nlp/text_data.py

```diff
@@ -0,0 +1,583 @@
+
+    @property
+    def is_multilabel(self) -> bool:
+        """Return True if label is multilabel.
+
+        Returns
+        -------
+        bool
+            True if label is multilabel.
+        """
+        return self._is_multilabel
+
+    def has_label(self) -> bool:
+        """Return True if label was set.
+
+        Returns
+        -------
+        bool
+           True if label was set.
+        """
+        return self._has_label
+
+    @classmethod
+    def cast_to_dataset(cls, obj: t.Any) -> 'TextData':
+        """Verify Dataset or transform to Dataset.
+
+        Function verifies that provided value is a non-empty instance of Dataset,
+        otherwise raises an exception, but if the 'cast' flag is set to True it will
+        also try to transform provided value to the Dataset instance.
+
+        Parameters
+        ----------
+        obj
+            value to verify
+
+        Raises
+        ------
+        DeepchecksValueError
+            if the provided value is not a TextData instance;
+            if the provided value cannot be transformed into Dataset instance;
+        """
+        if not isinstance(obj, cls):
+            raise DeepchecksValueError(f'{obj} is not a {cls.__name__} instance')
+        return obj
+
+    #     if isinstance(obj, pd.DataFrame):
+    #         get_logger().warning(
+    #             'Received a "pandas.DataFrame" instance. It is recommended to pass a "deepchecks.tabular.Dataset" '
+    #             'instance by initializing it with the data and metadata, '
+    #             'for example by doing "Dataset(dataframe, label=label, cat_features=cat_features)"'
+    #         )
+    #         obj = Dataset(obj)
+    #     elif not isinstance(obj, Dataset):
+    #         raise DeepchecksValueError(
+    #             f'non-empty instance of Dataset or DataFrame was expected, instead got {type(obj).__name__}'
+    #         )
+    #     return obj.copy(obj.data)
+
+    @classmethod
+    def datasets_share_label(cls, *datasets: 'TextData') -> bool:
+        """Verify that all provided datasets share same label name and task types.
+
+        Parameters
+        ----------
+        datasets : List[TextData]
+            list of TextData to validate
+
+        Returns
+        -------
+        bool
+            True if all TextData share same label names and task types, otherwise False
+
+        Raises
+        ------
+        AssertionError
+            'datasets' parameter is not a list;
+            'datasets' contains less than one dataset;
+        """
+        assert len(datasets) > 1, "'datasets' must contains at least two items"
+
+        label_name = datasets[0].label_name
+        task_type = datasets[0].task_type
+
+        for ds in datasets[1:]:
+            if ds.label_name != label_name:
+                return False
+            if ds.task_type != task_type:
+                return False
+
+        return True
+
+    def len_when_sampled(self, n_samples: int):
```

Comment:
Not sure this two methods add value, and if not to remove (ease of use of class later on)

Reply:
Used by context when adding sampling footnote